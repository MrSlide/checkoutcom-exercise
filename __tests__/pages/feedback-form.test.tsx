import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { fireEvent, render } from '@testing-library/react'
import { useRouter } from 'next/router'

import FeedbackFormPage from '../../src/pages'

jest.mock('next/router', function () {
  return {
    __esModule: true,
    useRouter: jest.fn(),
  }
})

describe('FeedbackFormPage component', function () {
  const mockRouterPush = jest.fn()
  const mockUseRouter = jest.mocked(useRouter)

  beforeEach(function () {
    mockUseRouter.mockReturnValue({
      push: mockRouterPush,
    } as any)
  })

  test('renders correctly', function () {
    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <FeedbackFormPage/>
    )

    expect(result).toMatchSnapshot()
  })
})
