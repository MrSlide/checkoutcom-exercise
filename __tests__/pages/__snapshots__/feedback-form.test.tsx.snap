// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackFormPage component renders correctly 1`] = `
<main>
  <Section>
    <Title>
      Feedback Form
    </Title>
    <Form
      onSubmit={[Function]}
    >
      <ResponsiveDualColLayout>
        <ResponsiveDualColLayoutColumn>
          <InputGroup
            label="Name"
          >
            <Input
              id="name-input"
              maxLength={100}
              name="name"
              required={true}
              validate={[Function]}
            />
          </InputGroup>
          <InputGroup
            label="Email"
          >
            <Input
              id="email-input"
              name="email"
              required={true}
              validate={[Function]}
            />
          </InputGroup>
          <InputGroup
            label="Rating"
          >
            <Input
              id="rating-input"
              max={5}
              maxLength={1}
              min={1}
              minLength={1}
              name="rating"
              required={true}
              step={1}
              type="number"
              validate={[Function]}
            />
          </InputGroup>
        </ResponsiveDualColLayoutColumn>
        <ResponsiveDualColLayoutColumn>
          <InputGroup
            label="Comment"
          >
            <Input
              id="comment-input"
              maxLength={1000}
              name="comment"
              required={true}
              type="text-area"
              validate={[Function]}
            />
          </InputGroup>
          <ButtonGroup>
            <SubmitButton>
              Submit
            </SubmitButton>
          </ButtonGroup>
        </ResponsiveDualColLayoutColumn>
      </ResponsiveDualColLayout>
    </Form>
  </Section>
</main>
`;
